x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/list_common/search_bar/search_bar.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import type { FC } from 'react';
+
+import type { SearchFilterConfig } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem, EuiButton, EuiSearchBar } from '@elastic/eui';
+import type { ExceptionListTypeEnum } from '@kbn/securitysolution-io-ts-list-types';
+import type { GetExceptionItemProps } from '../types';
+
+const ITEMS_SCHEMA = {
+  strict: true,
+  fields: {
+    created_by: {
+      type: 'string',
+    },
+    description: {
+      type: 'string',
+    },
+    id: {
+      type: 'string',
+    },
+    item_id: {
+      type: 'string',
+    },
+    list_id: {
+      type: 'string',
+    },
+    name: {
+      type: 'string',
+    },
+    os_types: {
+      type: 'string',
+    },
+    tags: {
+      type: 'string',
+    },
+  },
+};
+interface SearchBarProps {
+  addExceptionButtonText?: string;
+  placeholdertext?: string;
+  canAddException?: boolean; // TODO what is the default value
+
+  // TODO: REFACTOR: not to send the listType and handle it in the Parent
+  // Exception list type used to determine what type of item is
+  // being created when "onAddExceptionClick" is invoked
+  listType: ExceptionListTypeEnum;
+  isSearching?: boolean;
+  searchBarDataTestSubj?: string;
+  buttonDataTestSubj?: string;
```

Comment:
Same as suggested in one of the components above, about just having one `dataTestSubj` prop and then plugging it in to `data-test-subj={`${dataTestSubj}-component`

Reply:
#140985